src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+
+            # Compute the bounded footprint and recursively compute the intersection
+            bounded_footprint = other.approxBoundFootprint(centerZ, mesh_height)
+
+            return self.intersect(bounded_footprint)
+
+        if isinstance(other, PolygonalRegion):
+            # Other region can be represented by a polygon. We can slice the volume at the polygon's height,
+            # and then take the intersection of the resulting polygons.
+            origin_point = (self.mesh.centroid[0], self.mesh.centroid[1], other.z)
+            slice_3d = self.mesh.section(plane_origin=origin_point, plane_normal=[0,0,1])
+
+            if slice_3d is None:
+                return nowhere
+
+            slice_2d, _ = slice_3d.to_planar(to_2D=numpy.eye(4))
+            polygons = MultiPolygon(slice_2d.polygons_full) & other.polygons
+
+            if isinstance(polygons, shapely.geometry.Polygon):
+                polygons = MultiPolygon([polygons])
+
+            if polygons.is_empty:
+                return nowhere
+
+            return PolygonalRegion(polygon=polygons, z=other.z)
+
+        if isinstance(other, PathRegion):
+            # Extract lines from region
+            edges = [(other.vert_to_vec[v1], other.vert_to_vec[v2]) for v1,v2 in other.edges]
+
+            # Split lines anytime they cross the mesh boundaries
+            refined_polylines = []
+
+            for line_iter, line in enumerate(edges):
+                source, dest = line
+
+                ray = dest - source
+                ray = ray/numpy.linalg.norm(ray)
+
+                intersections = self.mesh.ray.intersects_location(ray_origins=[source],ray_directions=[ray])[0]
+
+                inner_points = sorted(intersections, key=lambda pos: numpy.linalg.norm(source-pos))
+                inner_points = filter(lambda point: numpy.linalg.norm(point-source) < numpy.linalg.norm(dest-source), inner_points)
+
+                refined_points = [source] + list(inner_points) + [dest]
+
+                refined_polylines.append(refined_points)
+
+            # Keep only lines and vertices for line segments in the mesh. 
+            internal_lines = []
+
+            for polyline in refined_polylines:
+                source = polyline[0]
+
+                for pt_iter in range(1, len(polyline)):
+                    dest = polyline[pt_iter]
+
+                    midpoint = (source + dest)/2
+                    if self.containsPoint(midpoint):
+                        internal_lines.append((source, dest))
+
+                    source = dest
+
+            # Check if merged lines is empty. If so, return the EmptyRegion. Otherwise, 
+            # transform merged lines back into a path region.
+            if internal_lines:
+                return PathRegion(polylines=internal_lines)
+            else:
+                return EmptyRegion("Empty mesh-polyline merge")
+
+        if isinstance(other, PolylineRegion):
```

Comment:
Much of this case is identical to `PathRegion` and should be consolidated.

Reply:
Good point, added to list.